Docs for security caveats / ownership claim of hostnames and subdomains.

ramr opened the pull request:
Finish up docs for https://trello.com/c/cyY7odgA/366-complete-up-docs-for-wildcard-subdomain-support-add-security-implications-and-add-more-help-text-to-the-oadm-router-command

@ahardin-rh / @knobunc  PTAL Thx

There's too many examples (and a lot of copy-pasta so would appreciate another set of eyes looking at this - thx) but hopefully they are clear ... 

For 3.4 only.

knobunc commented on install_config/router/default_haproxy_router.adoc:
```diff
@@ -907,6 +907,195 @@ This example uses `curl` with a local resolver to simulate the DNS lookup:
+associated with a wildcard route.
+
+Prior to wildcard routes, ownership was based on the claims made for a
+host name with the oldest route winning against any other claimants.
```
"with the oldest route" => "with the namespace with the oldest route"

ramr replied:
ack

knobunc commented on install_config/router/default_haproxy_router.adoc:
```diff
@@ -907,6 +907,195 @@ This example uses `curl` with a local resolver to simulate the DNS lookup:
+In addition, routes in other namespaces were allowed to claim
+non-overlapping hostnames.
+E.g. route `rone` in namespace `ns1` could claim `www.example.test` and
+   another route `rtwo` in namespace `d2` could claim `c3po.example.test`).
```
I see what you did there :-)

knobunc commented on install_config/router/default_haproxy_router.adoc:
```diff
@@ -907,6 +907,195 @@ This example uses `curl` with a local resolver to simulate the DNS lookup:
+   another route `rtwo` in namespace `d2` could claim `c3po.example.test`).
+
+This is still the case if there are _NO_ wildcard routes claiming that same
+subdomain (`*.example.test` in the above example).
```
Do we want to say *. there?  The subdomain is just example.test.

Or am I being too pedantic?

knobunc commented on install_config/router/default_haproxy_router.adoc:
```diff
@@ -907,6 +907,195 @@ This example uses `curl` with a local resolver to simulate the DNS lookup:
+subdomain (`*.example.test` in the above example).
+
+A wildcard route will however need to claim all the host names within a
+subdomain (`*.example.test` in the above example). A wildcard route's claim
```
Same

knobunc commented on install_config/router/default_haproxy_router.adoc:
```diff
@@ -907,6 +907,195 @@ This example uses `curl` with a local resolver to simulate the DNS lookup:
+E.g. if the oldest route `eldest` in namespace `ns1` making a claim for
+     subdomain `example.test` is a route which claims hostname
+     `owner.example.test`, then only a route in the `ns1` namespace will
+     be allowed to claim a wildcard route for that subdomain (`*.example.test`).
```
Same

knobunc replied:
I think we need to be clear that when the `eldest` route is deleted, then all bets are off and whatever namespace has the next oldest route wins the claim.

ramr replied:
and that should be route `wildcard` not eldest probably to be clearer - sounds a bit odd on a second read.

knobunc commented on install_config/router/default_haproxy_router.adoc:
```diff
@@ -907,6 +907,195 @@ This example uses `curl` with a local resolver to simulate the DNS lookup:
+----
+
+
+In the example below, a router that allows wildcard routes will not allow
```
Why does wildcard make a difference here?  Won't a non-wildcard router also forbid it?

ramr commented:
@knobunc  addressed your comments. I added a new example in the end to cover ownership changes within and across namespaces as putting it in the note section was making it unwieldy. PTAL thx

ramr commented:
squashed the commits.

ahardin-rh commented:
Merging and following up with minor style edits in a separate PR. Thanks!

ahardin-rh commented:
[rev_history]
|xref:../install_config/router/default_haproxy_router.adoc#install-config-router-default-haproxy[Setting up a Router -> Using the Default HAProxy Router]
|Added information about security caveats and ownership claims of host names and subdomains.
%
